src/migration/PoC_dfusion.js

```diff
@@ -11,48 +11,34 @@ async function migrate({ artifacts, deployer, network, account, web3, maxTokens
       web3,
     })
   }
-  const TokenOWLProxy = getDependency(
+  const feeToken = await getDeployedDependency(
     artifacts,
-    network,
     deployer,
     account,
     "@gnosis.pm/owl-token/build/contracts/TokenOWLProxy"
   )
-  const fee_token = await TokenOWLProxy.deployed()
 
-  const BatchExchange = getDependency(
+  const BatchExchange = getArtifact(artifacts, deployer, account, "@gnosis.pm/dex-contracts/build/contracts/BatchExchange")
+  const biMap = await getDeployedDependency(
     artifacts,
-    network,
-    deployer,
-    account,
-    "@gnosis.pm/dex-contracts/build/contracts/BatchExchange"
-  )
-  const BiMap = getDependency(
-    artifacts,
-    network,
     deployer,
     account,
     "@gnosis.pm/solidity-data-structures/build/contracts/IdToAddressBiMap"
   )
-  const IterableAppendOnlySet = getDependency(
+  const iterableAppendOnlySet = await getDeployedDependency(
     artifacts,
-    network,
     deployer,
     account,
     "@gnosis.pm/solidity-data-structures/build/contracts/IterableAppendOnlySet"
   )
 
-  // Hack to populate truffle artifact data correctly for linked libraries.
-  await BiMap.deployed()
-  await IterableAppendOnlySet.deployed()
-
   //linking libraries
-  await deployer.link(BiMap, BatchExchange)
-  await deployer.link(IterableAppendOnlySet, BatchExchange)
+  await deployer.link(biMap, BatchExchange)
+  await deployer.link(iterableAppendOnlySet, BatchExchange)
 
   // eslint-disable-next-line no-console
   console.log("Deploy BatchExchange contract")
-  await deployer.deploy(BatchExchange, maxTokens, fee_token.address)
+  await deployer.deploy(BatchExchange, maxTokens, feeToken.address)
```

Comment:
So an external project would still deploy a new instance of BatchExchange? How would they be able to use the already deployed one?